src/node_jsrtapi.cc

```diff
@@ -265,60 +179,51 @@ _Ret_maybenull_ JsValueRef CALLBACK CallbackWrapper(JsValueRef callee, bool isCo
   JsErrorCode error = JsNoError;
   JsValueRef undefinedValue;
   error = JsGetUndefinedValue(&undefinedValue);
-  CbInfo cbInfo;
-  cbInfo._callee = callee;
-  cbInfo._isConstructCall = isConstructCall;
-  cbInfo._argumentCount = argumentCount;
-  cbInfo._arguments = arguments;
+  CallbackInfo cbInfo;
+  cbInfo.isConstructCall = isConstructCall;
+  cbInfo.argc = argumentCount;
+  cbInfo.argv = reinterpret_cast<napi_value*>(arguments);
+  error = JsGetExternalData(callee, &cbInfo.data);
   cbInfo.returnValue = reinterpret_cast<napi_value>(undefinedValue);
   napi_callback cb = reinterpret_cast<napi_callback>(callbackState);
   // TODO(tawoll): get environment pointer instead of nullptr?
-  cb(nullptr, reinterpret_cast<napi_func_cb_info>(&cbInfo));
+  cb(nullptr, reinterpret_cast<napi_callback_info>(&cbInfo));
   return reinterpret_cast<JsValueRef>(cbInfo.returnValue);
 }
 
-napi_value napi_create_function(napi_env e, napi_callback cb) {
-  JsErrorCode error;
+napi_value napi_create_function(napi_env e, napi_callback cb, void* data) {
   JsValueRef function;
-  error = JsCreateFunction(CallbackWrapper, (void*)cb, &function);
-  return reinterpret_cast<napi_value>(function);
-}
+  JsCreateFunction(CallbackWrapper, (void*)cb, &function);
+
+  if (data) {
+    JsSetExternalData(function, data);
+  }
 
-napi_value napi_create_constructor_for_wrap(napi_env e, napi_callback cb) {
-  // TODO(tawoll): evaluate if we need to do something special here (I don't think so)
-  return napi_create_function(e, cb);
+  return reinterpret_cast<napi_value>(function);
 }
 
-napi_value napi_create_constructor_for_wrap_with_methods(
-  napi_env e,
-  napi_callback cb,
-  char* utf8name,
-  int methodcount,
-  napi_method_descriptor* methods) {
+napi_value napi_create_constructor(napi_env e, const char* utf8name, napi_callback cb,
+  void* data, int property_count, napi_property_descriptor* properties) {
+  // TODO: set properties
```

Comment:
The latest code in the V8 branches doesn't have this comment. The version I copied here is just a little behind since it was from last week when I created this PR. But it doesn't really matter since this file isn't compiled in this branch.